### PR TITLE
add max pagelen of 100 to /refs/branches call

### LINKED ISSUE #276: BitbucketCloudApiClient.executeMethod hangs the build
### Your checklist for this issue

- [x] Jenkins version

- [x] Plugin version

- [x] Bitbucket cloud

- [ ] Bitbucket server and version

### Description

Jenkins: 2.217
Plugin: 2.6.0
Bitbucket: cloud

Since a couple of days ago almost all our builds hang before even printing anything. 

Screen Shot 2020-01-27 at 5 15 39 PM

The only thing I could find initially was when I cancel the hanged build it shows this message:
```
Branch event
java.lang.InterruptedException: sleep interrupted
	at java.lang.Thread.sleep(Native Method)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.executeMethod(BitbucketCloudApiClient.java:858)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.getRequestAsInputStream(BitbucketCloudApiClient.java:885)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.getRequest(BitbucketCloudApiClient.java:909)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.getAllBranches(BitbucketCloudApiClient.java:1029)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.getBranchesByRef(BitbucketCloudApiClient.java:484)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.getBranches(BitbucketCloudApiClient.java:474)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMSource.retrieve(BitbucketSCMSource.java:791)
	at jenkins.scm.api.SCMSource.fetch(SCMSource.java:582)
	at org.jenkinsci.plugins.workflow.multibranch.SCMBinder.create(SCMBinder.java:98)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:299)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:428)
Finished: FAILURE
```

Screen Shot 2020-01-27 at 5 16 03 PM

Then I added a Logger for `com.cloudbees.jenkins.plugins.bitbucket` and it keeps printing this line over and over 
```
Jan 27, 2020 11:10:42 PM FINE com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient
Bitbucket Cloud API rate limit reached, sleeping for 5 sec then retry...
```

To me, it seems like the build is hanging on trying to get the Jenkinsfile from bitbucket. Not sure why we suddenly reached the limit, because we are not building more PR than usual. Could Bitbucket now consider retry as a request toward the limit?

### PR DESCRIPTION
**Pull request by Twb3**
An attempt to prevent hitting bitbucket rate limits on 2.0 api.  Default call to /refs/branches pagelen was 10, so increased to max of 100. 

fixes #276 

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by casz** [comment]
🤔 Was not aware your email can be verified on multiple accounts.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java [hunk 1]
```diff
@@ -478,6 +478,7 @@ public String getDefaultBranch() throws IOException, InterruptedException {
         String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + nodePath)
```
Comment (dblanchette) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: I think you need to add the parameter to the template like so:
> String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + nodePath + "{?pagelen}")

I tested locally with logging the URL

Comment (casz) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: Good catch 😅

Comment (casz) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: @dblanchette protip you can use suggestions:

```suggestion
        String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + nodePath + "{?pagelen}")
```

````
```suggestion
        String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + nodePath + "{?pagelen}")
```
````

Comment (dblanchette) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java]: Thanks for the tip!

**Comment by Twb3** [comment]
@dblanchette Good catch! thank you

**Pull request merged**